cyclonedx/model/component.py

```diff
@@ -1159,6 +1159,16 @@ def has_vulnerabilities(self) -> bool:
         """
         return bool(self.get_vulnerabilities())
 
+    def get_all_nested_components(self, include_self: bool = False) -> Set["Component"]:
```

Comment:
are there any unit tests for this?

i have the idea that this function is not working as expected, when param  `include_self` is set to `False`.
Code looks like it would return an empty set, then - no matther how deep the nesting actually is

Reply:
@jkowalleck - there was limited coverage in the output tests, but have now added some specific tests to cover this.

It appears to be working as intended.